### PR TITLE
pt012 rule enabled amazon

### PR DESCRIPTION
**Pull request by Bowrna**
related: #38270
fix PT012 checks for amazon.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/providers/amazon/aws/hooks/test_base_aws.py [hunk 1]
```diff
@@ -1123,13 +1123,13 @@ def test_raise_no_creds_default_credentials_strategy(tmp_path_factory, monkeypat
...
-    with pytest.raises(NoCredentialsError):
+    with pytest.raises(NoCredentialsError) as credential_error:
         # Call AWS STS API method GetCallerIdentity
         # which should return result in case of valid credentials
-        result = hook.conn.get_caller_identity()
+        hook.conn.get_caller_identity()
         # In normal circumstances lines below should not execute.
         # We want to show additional information why this test not passed
-        assert not result, f"Credentials Method: {hook.get_session().get_credentials().method}"
+    assert str(credential_error.value) == "Unable to locate credentials"
```
Comment (Taragolis) [inline tests/providers/amazon/aws/hooks/test_base_aws.py]: The commend no more relevant, so it should be changed / removed

File: pyproject.toml [hunk 2]
```diff
@@ -362,20 +362,20 @@ combine-as-imports = true
...
+#"tests/providers/amazon/aws/hooks/test_s3.py" = ["PT012"]
+#"tests/providers/amazon/aws/operators/test_emr_serverless.py" = ["PT012"]
+#"tests/providers/amazon/aws/operators/test_redshift_data.py" = ["PT012"]
+#"tests/providers/amazon/aws/sensors/test_glacier.py" = ["PT012"]
+#"tests/providers/amazon/aws/sensors/test_glue.py" = ["PT012"]
+#"tests/providers/amazon/aws/sensors/test_lambda_function.py" = ["PT012"]
+#"tests/providers/amazon/aws/system/utils/test_helpers.py" = ["PT012"]
+#"tests/providers/amazon/aws/transfers/test_redshift_to_s3.py" = ["PT012"]
+#"tests/providers/amazon/aws/triggers/test_ecs.py" = ["PT012"]
+#"tests/providers/amazon/aws/waiters/test_neptune.py" = ["PT012"]
```
Comment (Taragolis) [inline pyproject.toml]:
```suggestion
```

**Review by potiuk — APPROVED** [review]

File: tests/providers/amazon/aws/hooks/test_base_aws.py (hunk 1 above)
Comment (potiuk) [reply tests/providers/amazon/aws/hooks/test_base_aws.py]: yep. just that one nit to complete.

Comment (Bowrna) [reply tests/providers/amazon/aws/hooks/test_base_aws.py]: yes I have removed the comment. @Taragolis @potiuk

**Review by Taragolis — APPROVED** [review]

**Pull request merged**